protocols/floodsub/src/layer.rs

```diff
@@ -161,17 +161,20 @@ impl<TSubstream> Floodsub<TSubstream> {
         true
     }
 
-    /// Publishes a message to the network.
-    ///
-    /// > **Note**: Doesn't do anything if we're not subscribed to the topic.
+    /// Publishes a message to the network, if we're subscribed to the topic only.
     pub fn publish(&mut self, topic: impl Into<TopicHash>, data: impl Into<Vec<u8>>) {
-        self.publish_many(iter::once(topic), data)
+        self.publish_many(iter::once(topic), data, true)
+    }
+
+    /// Publishes a message to the network, even if we're not subscribed to the topic.
+    pub fn publish_any(&mut self, topic: impl Into<TopicHash>, data: impl Into<Vec<u8>>) {
+        self.publish_many(iter::once(topic), data, false)
     }
 
     /// Publishes a message with multiple topics to the network.
     ///
-    /// > **Note**: Doesn't do anything if we're not subscribed to any of the topics.
-    pub fn publish_many(&mut self, topic: impl IntoIterator<Item = impl Into<TopicHash>>, data: impl Into<Vec<u8>>) {
+    /// > **Note**: By enabling `check_self_subscriptions`, message will be sent to the network, if we are subscribed to the topic only.
+    pub fn publish_many(&mut self, topic: impl IntoIterator<Item = impl Into<TopicHash>>, data: impl Into<Vec<u8>>, check_self_subscriptions: bool) {
```

Comment:
I think it should be a new method, named something like `publish_many_inner`, instead of a modification of the existing method.